### DISCUSSION TITLE
Authorization header missing, in seemingly random cases

### DISCUSSION THREAD

**Discussion by marqhuez** [pocketbase/pocketbase#1514] [Q&A]
Hey!

I have a weird problem. I have a login feature implemented in my application, and I have a board, which selects all the records that belongs to the logged in user. But seemingly random, sometimes the records are empty. I have checked the requests, and before every request to the collection, there is a Preflight request, with OPTION method, and after this request, there is another, which actually selects the records. When there are no records in the result, there is no Preflight request, and on the actual request for the data doesn't contain the Authorization header, so I guess pb doesn't know what user is requesting the data.
I check for user logged in status before selection all records, and there is a model, with user id, everything seems correct there.
After a page load, everything works fine.

The app runs on Linode.

Is there something I am doing wrong, or is this a bug, or maybe I have to config something in Linode, and it has nothing to do with pb?

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand your issue.

The preflight request is send by the browsers for CORS (it is a "safety" measure to ensure that the request being sent is trusted/allowed by the server) and they usually don't have body.

If you are getting a list result with empty `items` key in the response, then this most likely means that your "List/Search API rule" is not satisfied by the client (eg. is not authorized; we don't return 403 in this cases because the API rules act also as filters).

**Reply by marqhuez** [reply]
I have an example here:

![image](https://user-images.githubusercontent.com/80782266/211194048-6919cae9-c32a-4790-ad88-091f175db04e.png)
Here I have this 2 requests.
The first one is the Preflight request, after that there is the actual GET request, to fetch the collection. This is the correct behavior.

And here, I only have one GET request to fetch data, there is no Preflight request.
![image](https://user-images.githubusercontent.com/80782266/211194032-28be2e70-3fce-4c68-8c7b-4e1b5a870c27.png)
In this case, in the GET request there is no Authorization header, and the items are empty.

The request payload is the same, the request headers are the same, except for the Authorization.

The List/Search API rule is set up like this:
user = @request.auth.id

Both requests are sent in the exact same scenario after I log in, I get redirected to a page, where I fetch data. I use the same user every time. At the time of fetching the data, there is always a logged in user, so the pb client should have the user id as well.
The app is written in SvelteKit by the way.

I hope this clears things a bit.

**Reply by ganigeorgiev** [reply]
I'm sorry, I'm still not sure that I understand the issue.

If you have authenticated with the JS SDK (eg. with `authWithPassword()`) and not seeing an `Authorization` header being sent with the request, then this could be caused by 2 reasons:
- The token exp claim has expired (you can check it with `pb.authStore.isValid` right before sending the `getList()` request)
- Client-side issue (eg. your code initializes different `PocketBase` js sdk instances)

Without code sample how you are sending the request and where your code is executed (browser, node server-endpoint, mixed) it is difficult to pinpoint the exact reason for the above behavior.
If you can provide a minimal reproducible repo I can give it a try and investigate it, but I don't think this is an issue with PocketBase or the JS SDK.
